daemon/config.go

```diff
@@ -218,6 +239,28 @@ func getConflictFreeConfiguration(configFile string, flags *flag.FlagSet) (*Conf
 		return nil, err
 	}
 
+	// Sanitize values of null so that configurations like
+	// `{ "max-downloads" : null }` will not be considered as 0.
+	// Note: Only the first level is checked.
```

Comment:
I think a better way to do this is to make `MaxDownloads` and `MaxUploads` `*int` instead of `int`. They will be nil pointers unless the value can be unmarshalled as an integer.